cmd/gitcloner/integrationtests/clone_test.go

```diff
@@ -0,0 +1,465 @@
+			url, err := getHTTPSURL(context.Background(), container)
+			Expect(err).NotTo(HaveOccurred())
+			err = createRepo(url, "assets/repo", private)
+			Expect(err).NotTo(HaveOccurred())
+			opts.Repo = url + "/test/test-repo"
+			tmp = createTempFolder()
+			opts.Path = tmp
+		})
+
+		JustAfterEach(func() {
+			err := os.RemoveAll(tmp)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		When("Public repo that contains a README.md file", func() {
+			BeforeEach(func() {
+				private = false
+				opts = &cmd.Options{
+					InsecureSkipTLS: true,
+				}
+			})
+
+			JustBeforeEach(func() {
+				c := gogit.NewCloner()
+				cloneErr := c.CloneRepo(opts)
+				Expect(cloneErr).NotTo(HaveOccurred())
+			})
+
+			It("README.md file is present", func() {
+				_, err := os.Stat(tmp + "/README.md")
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+
+		When("Private repo that contains a README.md file", func() {
+			When("No authentication is provided", func() {
+				BeforeEach(func() {
+					private = true
+					opts = &cmd.Options{
+						InsecureSkipTLS: true,
+					}
+				})
+
+				JustBeforeEach(func() {
+					c := gogit.NewCloner()
+					cloneErr = c.CloneRepo(opts)
+				})
+
+				It("Fails to clone the repo", func() {
+					Expect(cloneErr.Error()).To(Equal("authentication required"))
+				})
+			})
+
+			When("Basic authentication is provided", func() {
+				BeforeEach(func() {
+					private = true
+					opts = &cmd.Options{
+						InsecureSkipTLS: true,
+						Username:        gogsUser,
+						PasswordFile:    "assets/gogs/password",
+					}
+				})
+
+				JustBeforeEach(func() {
+					c := gogit.NewCloner()
+					cloneErr = c.CloneRepo(opts)
+					Expect(cloneErr).NotTo(HaveOccurred())
+				})
+
+				It("README.md file is present", func() {
+					_, err := os.Stat(tmp + "/README.md")
+					Expect(err).NotTo(HaveOccurred())
+				})
+			})
+		})
+
+		When("Insecure skip tls is not true, and no caBundle is provided", func() {
+			BeforeEach(func() {
+				private = false
+				opts = &cmd.Options{}
+			})
+
+			JustBeforeEach(func() {
+				c := gogit.NewCloner()
+				cloneErr = c.CloneRepo(opts)
+			})
+
+			It("Fails to clone the repo", func() {
+				Expect(cloneErr.Error()).To(ContainSubstring("x509: certificate signed by unknown authority"))
+			})
+		})
+
+		When("Insecure skip tls is not true, and caBundle is provided", func() {
+			var tmpFile string
+
+			BeforeEach(func() {
+				private = false
+				opts = &cmd.Options{}
+			})
+
+			AfterEach(func() {
+				err := os.Remove(tmpFile)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			JustBeforeEach(func() {
+				caBundleFile, err := os.CreateTemp("", "gitcloner")
+				Expect(err).NotTo(HaveOccurred())
+				tmpFile = caBundleFile.Name()
+				_, err = caBundleFile.Write(gogsCABundle)
+				Expect(err).NotTo(HaveOccurred())
+				opts.CABundleFile = caBundleFile.Name()
+
+				c := gogit.NewCloner()
+				cloneErr = c.CloneRepo(opts)
+			})
+
+			It("README.md file is present", func() {
```

Comment:
Same comment as on L81.

Reply:
done in https://github.com/rancher/gitjob/commit/43c66e7e40a9791d03bff8bc45172e4d1e87742e